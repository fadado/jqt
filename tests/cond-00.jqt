=======================
Conditional blocks
=======================

_____
{% .yes//empty %}
	I expand!
{% end %}
{% .string//empty %}
	I expand to '{{.}}'
{% end %}
{% .number//empty %}
	I expand to '{{.}}'
{%end%}
{% .number//empty | $M %}
	I expand preserving $M: '{{$M.number}}'
{% end %}
_____
{% .not//empty %}
	I vanish!
{% end %}
{% .nil//empty %}
	I vanish!
{% end %}
_____

{% .yes//empty | $M %}
	{% .string//empty %}I expand to '{{.}}'
{% end %}
{% .yes//empty | $M %}
	{% .string//empty %}
I expand to '{{.}}'
	{% end %}
{% end %}

{% (.number > 4)//empty %}
	then not
{% end %}
{% (.number < 3 | not)//empty %}
	else yes
{% end %}

{% (.number < 1)//empty %}
	case 1
{% end %}
{% (.number < 1 | not)//empty | $M %}
{% (.number < 2)//empty %}
	case 2
{% end %}
{% (.number < 2 | not)//empty %}
	case else
{% end %}
{% end %}
