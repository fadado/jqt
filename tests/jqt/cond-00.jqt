=======================
Conditional blocks
=======================

_____
{% .yes//empty %}
	I expand!
{% end %}
{% .string//empty %}
	I expand to '{{.}}'
{% end %}
{% .number//empty %}
	I expand to '{{.}}'
{%end%}
{% .number//empty | $jqt %}
	I expand preserving $jqt: '{{$jqt.number}}'
{% end %}
_____
{% .not//empty %}
	I vanish!
{% end %}
{% .nil//empty %}
	I vanish!
{% end %}
_____

{% .yes//empty | $jqt %}
	{% .string//empty %}I expand to '{{.}}'
{% end %}
{% .yes//empty | $jqt %}
	{% .string//empty %}
I expand to '{{.}}'
	{% end %}
{% end %}

{% (.number > 4)//empty %}
	then not
{% end %}
{% (.number < 3 | not)//empty %}
	else yes
{% end %}

{% (.number < 1)//empty %}
	case 1
{% end %}
{% (.number < 1 | not)//empty | $jqt %}
{% (.number < 2)//empty %}
	case 2
{% end %}
{% (.number < 2 | not)//empty %}
	case else
{% end %}
{% end %}
