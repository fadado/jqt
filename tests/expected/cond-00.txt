=======================
Conditional blocks
=======================

_____
	I expand!
	I expand to 'Neque porro quisquam'
	I expand to '3.141592653589793'
	I expand preserving $M: '3.141592653589793'
_____
_____

I expand to 'Neque porro quisquam'
I expand to 'Neque porro quisquam'

	else yes

	case else
